assets/js/frontend/components/review/index.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import ReadMore from '../read-more';
+
+function getReviewClasses( isLoading ) {
+	const classArray = [ 'wc-block-reviews-by-product__item' ];
+
+	if ( isLoading ) {
+		classArray.push( 'is-loading' );
+	}
+
+	return classArray.join( ' ' );
+}
+
+function getReviewImage( review, imageType, isLoading ) {
+	if ( isLoading || ! review ) {
+		return (
+			<div className="wc-block-reviews-by-product__image" width="48" height="48" />
+		);
+	}
+
+	return (
+		<div className="wc-block-reviews-by-product__image">
+			{ imageType === 'product' ? (
+				<img aria-hidden="true" alt="" src={ review.product_picture } className="wc-block-reviews-by-product__image" width="48" height="48" />
+			) : (
+				<img aria-hidden="true" alt="" src={ review.reviewer_avatar_urls[ '48' ] } srcSet={ review.reviewer_avatar_urls[ '96' ] + ' 2x' } className="wc-block-reviews-by-product__image" width="48" height="48" />
+			) }
+			{ review.verified && (
+				<div className="wc-block-reviews-by-product__verified" title={ __( 'Verified buyer', 'woo-gutenberg-products-block' ) }>{ __( 'Verified buyer', 'woo-gutenberg-products-block' ) }</div>
+			) }
+		</div>
+	);
+}
+
+function getReviewContent( review ) {
+	return (
+		<ReadMore
+			maxLines={ 10 }
+			moreText={ __( 'Read full review', 'woo-gutenberg-products-block' ) }
+			lessText={ __( 'Hide full review', 'woo-gutenberg-products-block' ) }
+			className="wc-block-reviews-by-product__text"
+		>
+			<div
+				dangerouslySetInnerHTML={ {
+					// `content` is the `review` parameter returned by the `reviews` endpoint.
+					// It's filtered with `wp_filter_post_kses()`, which removes dangerous HTML tags,
+					// so using it inside `dangerouslySetInnerHTML` is safe.
+					__html: review.review || '',
+				} }
+			/>
+		</ReadMore>
+	);
+}
+
+const Review = ( { attributes, review = {} } ) => {
```

Comment:
Let's name this ReviewListItem

Reply:
Also done in fde9e972de5bba44d64fe736d72f15705bb0de47.